_applications/equoid.adoc

```diff
@@ -0,0 +1,219 @@
+* An OpenShift project you own.
+
+[[procedure]]
+=== Procedure
+
+. Install Image Streams and Templates
++
+Depending on your OpenShift instance, some of the image streams necessary for Equoid may not be available. To be certain you have what's necessary to build Equoid, run the following sequence of commands:
++
+....
+oc create -f https://raw.githubusercontent.com/radanalyticsio/equoid-openshift/master/fabric8-image-streams.json
+oc create -f https://radanalytics.io/resources.yaml
+oc create -f https://raw.githubusercontent.com/infinispan/infinispan-openshift-templates/master/templates/infinispan-ephemeral.json
+....
++
+These provide the Fabric8 fuse-java image stream for the data-publisher service, the radanalytics.io Oshinko resources for facilitating Spark cluster creation and deployment as well as S2I for the data-handler, and the Infinispan template for the caching microservice, respectively. 
+
+. Launch Artemis
++
+We'll get Artemis up and running, as it is necessary for both the publisher and handler:
++
+....
+oc create -f https://raw.githubusercontent.com/radanalyticsio/equoid-openshift/master/artemis-rc.yaml
+....
+
+. Launch Infinispan Cache
++
+We'll want our caching layer in place before we begin streaming and ingesting. The following will get an Infinispan pod running. `APPLICATION_USER`, `APPLICATION_PASSWORD`, `MANAGEMENT_USER` and `MANAGEMENT_PASSWORD` should be changed to your desired values. 
++
+....
+oc new-app --template=infinispan-ephemeral \
+    -l app=datagrid \
+    -p APPLICATION_NAME=datagrid \
+    -p NAMESPACE=`oc project -q` \
+    -p APPLICATION_USER=<YOUR_APP_USERNAME> \
+    -p APPLICATION_PASSWORD=<YOUR_APP_PASSWORD> \
+    -p MANAGEMENT_USER=<YOUR_MGMT_USERNAME> \
+    -p MANAGEMENT_PASSWORD=<YOUR_MGMT_PASSWORD>
+....
++
+
+. Launch Data Publisher
++
+We can next begin publishing our streaming data by starting an instance of the data publisher. `OP_MODE` described later should be set to the mode you wish. Also, `DATA_URL_PRIMARY` should be set to the location of your data set or data potential values file. 
++
+....
+oc new-app \
+    -l app=publisher \
+    -e OP_MODE={single|dual|linear} \
+    -e DATA_URL_PRIMARY=<YOUR_DATASET_URL> \
+    --image-stream=`oc project -q`/fuse-java \
+    https://github.com/radanalyticsio/equoid-data-publisher
+....
+
+. Launch Data Handler
++
+We are now ready to start our data handler. Note that `INFINISPAN_HOST` and `INFINISPAN_PORT` need to be set to appropriate values, in the case of the template provided, these are `datagrid-hotrod` and `11222` respectively. In addition, `WINDOW_SECONDS` is the number of second for which you wish to track the top-k items, `SLIDE_SECONDS` represents the number of seconds to slide your data window each iteration, and `BATCH_SECONDS` is the size of samples you wish to acquire from the message stream at a time. Finally, `OP_MODE` needs to be set to the same value as set for the data publisher.
++
```

Comment:
i think the explanation of the environment options is great here, but i think we should provide some sane defaults in the code box.

Reply:
ACK, and included in next rev